### PR TITLE
i18n Translation ready

### PR DESCRIPTION
**Pull request by ipokkel**
i18n Translation Ready.

* Added Text Domain
* Add Domain Path
* Converted HTML text to translatable strings
* Converted text strings to translatable strings
* Changed text domain for pmprosed_plugin_row_meta from "pmpro" to "paid-memberships-pro"
* Added folder languages
* Create & Add pmpro-set-expiration-dates.pot in languages folder

### CODE REVIEW

File: pmpro-set-expiration-dates.php [hunk 1]
```diff
@@ -21,18 +29,16 @@ function pmprosed_pmpro_membership_level_after_other_settings()
     else
         $set_expiration_date = "";
     ?>
-    <h3 class="topborder">Set Expiration Date</h3>
-    <p>To have this level expire on a specific date, enter it below in YYYY-MM-DD format. <strong>Note:</strong> You
-        must also set an expiration date above (e.g. 1 Year) which will be overwritten by the value below.</p>
+    <h3 class="topborder"><?php _e( 'Set Expiration Date', 'pmpro-set-expiration-dates' ); ?></h3>
+    <p><?php _e( 'To have this level expire on a specific date, enter it below in YYYY-MM-DD format. <strong>Note:</strong> You must also set an expiration date above (e.g. 1 Year) which will be overwritten by the value below.', 'pmpro-set-expiration-dates' ); ?></p>
```
Comment (andrewlimaza) [inline pmpro-set-expiration-dates.php]: Try to avoid wrapping HTML elements inside i18n functions.

File: pmpro-set-expiration-dates.php [hunk 2]
```diff
@@ -298,8 +303,8 @@ function pmprosed_plugin_row_meta($links, $file)
 {
     if (strpos($file, 'pmpro-set-expiration-dates.php') !== false) {
         $set_links = array(
-            '<a href="' . esc_url('http://www.paidmembershipspro.com/add-ons/plugins-on-github/pmpro-expiration-date/') . '" title="' . esc_attr(__('View Documentation', 'pmpro')) . '">' . __('Docs', 'pmpro') . '</a>',
-            '<a href="' . esc_url('http://paidmembershipspro.com/support/') . '" title="' . esc_attr(__('Visit Customer Support Forum', 'pmpro')) . '">' . __('Support', 'pmpro') . '</a>',
+            '<a href="' . esc_url('http://www.paidmembershipspro.com/add-ons/plugins-on-github/pmpro-expiration-date/') . '" title="' . esc_attr(__('View Documentation', 'paid-memberships-pro')) . '">' . __('Docs', 'paid-memberships-pro') . '</a>',
```
Comment (andrewlimaza) [inline pmpro-set-expiration-dates.php]: I think the text domain should be same as text-domain of the plugin and not PMPro core.

File: pmpro-set-expiration-dates.php (hunk 1 above)
Comment (ipokkel) [reply pmpro-set-expiration-dates.php]: updated to printf with html elements as variables in string

File: pmpro-set-expiration-dates.php (hunk 2 above)
Comment (ipokkel) [reply pmpro-set-expiration-dates.php]: update text domain to that of plugin 'pmpro-set-expiration-dates'

File: pmpro-set-expiration-dates.php [hunk 3]
```diff
@@ -30,7 +30,11 @@ function pmprosed_pmpro_membership_level_after_other_settings()
         $set_expiration_date = "";
     ?>
     <h3 class="topborder"><?php _e( 'Set Expiration Date', 'pmpro-set-expiration-dates' ); ?></h3>
-    <p><?php _e( 'To have this level expire on a specific date, enter it below in YYYY-MM-DD format. <strong>Note:</strong> You must also set an expiration date above (e.g. 1 Year) which will be overwritten by the value below.', 'pmpro-set-expiration-dates' ); ?></p>
+    <p><?php printf( 
+        __( 'To have this level expire on a specific date, enter it below in YYYY-MM-DD format. %1$sNote:%2$s You must also set an expiration date above (e.g. 1 Year) which will be overwritten by the value below.', 'pmpro-set-expiration-dates' ),
```
Comment (andrewlimaza) [inline pmpro-set-expiration-dates.php]: I think concatenating is better here.

Something like this:

`echo __( 'To have this level expire on a specific date, enter it below in YYYY-MM-DD format., 'pmpro-set-expiration-dates') . '<strong>' . __( 'Note', 'pmpro-set-expiration-dates' ) . ':</strong>' . __( 'You must also set an expiration date above (e.g. 1 Year) which will be overwritten by the value below.', 'pmpro-set-expiration-dates' ),`

**Comment by ipokkel** [comment]
changed load text domain to init action

**Pull request merged**